piqueserver/player.py

```diff
@@ -403,16 +403,7 @@ def on_hack_attempt(self, reason):
         self.kick(reason)
 
     def on_user_login(self, user_type, verbose=True):
-        if user_type == 'admin':
-            self.admin = True
```

Comment:
Do any other parts of the code rely on this connection.admin attribute? It seems like they would.

Reply:
All the logic from on_user_login has been split into `set_user_type` and `notify_login` method/function.
```python
    def set_user_type(self, connection, user_type: str) -> None:
        if user_type == 'admin':
            connection.admin = True
            connection.speedhack_detect = False
        connection.user_types.add(user_type)
        rights = set(self.get_rights(user_type))
        connection.rights.update(rights)
```